File: OC.Domain/Models/Branches/BranchMinistry.cs
using OC.Domain.Interfaces;

namespace OC.Domain.Models.Branches
{
    public class BranchMinistry : AuditableEntity, IEntityBase
    {
        public BranchMinistry() { 
            BranchMinistryLeaderships= new HashSet<BranchMinistryLeadership>();
        }

        public long Id { get; set; }
        public long MinistryId { get; set;}
        public long BranchId { get; set; }

        // Relationships
        public virtual Branch Branch { get; set; }
        public virtual Ministry Ministry { get; set; }
        public virtual ICollection<BranchMinistryLeadership> BranchMinistryLeaderships { get; set; }
    }
}

File: OC.UI/src/app/blci/api/branches/branch-ministry.ts
import { Branch } from "./branch";
import { BranchMinistryLeadership } from "./branch-ministry-leadership";
import { Ministry } from "./ministry";

export interface BranchMinistry {
    id: number;
    ministryId: number;
    branchId: number;
    isActive: boolean;
    branch: Branch;
    ministry: Ministry;
    branchMinistryLeaderships: BranchMinistryLeadership[];
}
